packages/fury-adapter-oas3-parser/lib/parser/oas/parseOperationObject.js

```diff
@@ -33,16 +33,40 @@ function createTransactions(namespace, member, operation) {
   });
 }
 
+function hrefVariablesFromParameters(namespace, parameters) {
+  const path = parameters.get('path')
+    ? parameters.get('path')
+    : new namespace.elements.HrefVariables();
+
+  const query = parameters.get('query')
+    ? parameters.get('query')
+    : new namespace.elements.HrefVariables();
+
+  if (!path.isEmpty || !query.isEmpty) {
+    return path.concat(query);
+  }
+
+  return undefined;
```

Comment:
Should it be undefined or empty array? 🤔

Reply:
It should be undefined because then we won't set an empty array in the `attributes` for `hrefVariables`. If we do set undefined value in attributes it won't turn up in the result.

The line further down that makes use will behave as described:

```js
transition.hrefVariables = hrefVariablesFromParameters(namespace, parameters);
```

NOTE, if we do set an empty array when there are no parameters, we would then be overriding the parameters section of the path item in this particular operation which is not desired.

Reply:
Note, `transition.hrefVariables = undefined` will translate to a `transition.attributes.set('hrefVariables', undefined)`